tiny_dnn/layers/layer.h

```diff
@@ -378,15 +378,14 @@ class layer : public node {
    * @param trainable_only flag to return only the trainable parameters.
    * @return std::vector of pointers to parameters
    */
-  std::vector<parameter *> parameters(bool trainable_only = false) {
-    std::vector<parameter *> vp;
-    for (size_t i = 0; i < parameters_.size(); i++) {
-      if (!trainable_only ||
-          (parameters_[i]->is_trainable() && trainable_only)) {
-        vp.push_back(parameters_[i]);
+  Parameters parameters(bool trainable_only = false) {
+    Parameters parameters;
+    for (Parameter *parameter : parameters) {
```

Comment:
you can use auto: `for (const auto& parameter : parameters_)`

is more C++11 style but at the same time confusing if you don't know that parameters_ is a vector of pointers. At least it will work: http://cpp.sh/82rig